src/v/kafka/server/handlers/describe_producers.cc

```diff
@@ -34,11 +34,11 @@ namespace {
 
 using ret_res = checked<cluster::rm_stm::transaction_set, kafka::error_code>;
 
-int64_t
-convert_rm_stm_time_to_milliseconds(cluster::rm_stm::time_point_type time) {
-    auto diff = cluster::rm_stm::clock_type::now() - time;
-    auto now = std::chrono::steady_clock::now();
-    return now.time_since_epoch() / 1ms - diff / 1ms;
+static int64_t
+convert_rm_stm_time_to_ts_ms(cluster::rm_stm::time_point_type time) {
+    auto diff = (cluster::rm_stm::clock_type::now() - time) / 1ms;
```

Comment:
nit: I think it is more readable to use a duration_cast<millis> here and below.

Reply:
+1

Reply:
> nit: I think it is more readable to use a duration_cast here and below.

yeh it's usually better to cast at the end so that earlier calculations will preserve precision.

auto ts = std::chrono::system_clock::now().time_since_epoch() - (cluster::rm_stm::clock_type::now() - time);
return std::chrono::duration_cast<std::chrono::milliseconds>(ts);